sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventHubClient.cs

```diff
@@ -0,0 +1,421 @@
+        ///   is assumed that callers are trusted and have performed deep validation.
+        ///
+        ///   Any parameters passed are assumed to be owned by this instance and safe to mutate or dispose;
+        ///   creation of clones or otherwise protecting the parameters is assumed to be the purview of the
+        ///   caller.
+        /// </remarks>
+        ///
+        protected AmqpEventHubClient(string host,
+                                     string eventHubName,
+                                     TokenCredential credential,
+                                     EventHubClientOptions clientOptions,
+                                     EventHubRetryPolicy defaultRetryPolicy,
+                                     AmqpConnectionScope connectionScope,
+                                     AmqpMessageConverter messageConverter)
+        {
+            Guard.ArgumentNotNullOrEmpty(nameof(host), host);
+            Guard.ArgumentNotNullOrEmpty(nameof(eventHubName), eventHubName);
+            Guard.ArgumentNotNull(nameof(credential), credential);
+            Guard.ArgumentNotNull(nameof(clientOptions), clientOptions);
+            Guard.ArgumentNotNull(nameof(defaultRetryPolicy), defaultRetryPolicy);
+
+            try
+            {
+                EventHubsEventSource.Log.EventHubClientCreateStart(host, eventHubName);
+
+                _retryPolicy = defaultRetryPolicy;
+                _tryTimeout = _retryPolicy.CalculateTryTimeout(0);
+
+                EventHubName = eventHubName;
+                Credential = credential;
+                MessageConverter = messageConverter ?? new AmqpMessageConverter();
+
+                if (connectionScope == null)
+                {
+                    var endpointBuilder = new UriBuilder
+                    {
+                        Scheme = clientOptions.TransportType.GetUriScheme(),
+                        Host = host
+                    };
+
+                    connectionScope = new AmqpConnectionScope(endpointBuilder.Uri, eventHubName, credential, clientOptions.TransportType, clientOptions.Proxy);
+                }
+
+                ConnectionScope = connectionScope;
+                ManagementLink = new FaultTolerantAmqpObject<RequestResponseAmqpLink>(timeout => ConnectionScope.OpenManagementLinkAsync(timeout, CancellationToken.None), link => link.SafeClose());
+            }
+            finally
+            {
+                EventHubsEventSource.Log.EventHubClientCreateComplete(host, eventHubName);
+            }
+
+        }
+
+        /// <summary>
+        ///   Updates the active retry policy for the client.
+        /// </summary>
+        ///
+        /// <param name="newRetryPolicy">The retry policy to set as active.</param>
+        ///
+        public override void UpdateRetryPolicy(EventHubRetryPolicy newRetryPolicy)
+        {
+            _retryPolicy = newRetryPolicy;
```

Comment:
Why not performing a sanity check like in the constructor?

Reply:
Good catch!  That should definitely be there.